tests/test_markdown_components.py

```diff
@@ -0,0 +1,142 @@
+            ),
+            dag.AgGrid(
+                id="not_dangerous",
+                columnSize="sizeToFit",
+                columnDefs=columnDefs,
+                rowData=rowData,
+            ),
+            html.Hr(),
+        ]
+    )
+
+
+    raw_html_example2 = html.Div(
+        [
+            dcc.Markdown(
+                "This grid has both Markdown and raw HTML. `dangerously_allow_code=True` so the links render"
+            ),
+            dag.AgGrid(
+                id="dangerous",
+                columnSize="sizeToFit",
+                columnDefs=columnDefs_allow_html,
+                rowData=rowData,
+                dangerously_allow_code=True,
+            ),
+        ]
+    )
+
+
+    app.layout = html.Div(
+        [raw_html_example1, raw_html_example2],
+        style={"flexWrap": "wrap"},
+    )
+
+    dash_duo.start_server(app)
+
+    grid = utils.Grid(dash_duo, "not_dangerous")
+
+    grid.wait_for_cell_text(0, 0, "Toyota in italics")
+    assert grid.get_cell(0, 0).get_attribute(
+        'innerHTML') == '<div class="agGrid-Markdown"><div><em>Toyota</em> in italics</div></div>'
+    grid.wait_for_cell_text(1, 1, "Mondeo")
+    grid.wait_for_cell_text(2, 2, "Example")
+    assert grid.get_cell(2, 2).get_attribute('innerHTML') == '<div class="agGrid-Markdown"><div><a href="#" target="_blank">Example</a></div></div>'
+    grid.wait_for_cell_text(1, 2, '<a href="#" target="_blank">Link to new tab</a>')
+    assert grid.get_cell(2, 3).get_attribute('innerHTML') == '<div class="agGrid-Markdown">' \
+                                                             '<div><img src="https://www.ag-g' \
+                                                             'rid.com/example-assets/weather/rain.png" ' \
+                                                             'alt="alt text: rain"></div></div>'
+
+    grid = utils.Grid(dash_duo, "dangerous")
+
+    grid.wait_for_cell_text(0, 0, "Toyota in italics")
+    assert grid.get_cell(0, 0).get_attribute(
+        'innerHTML') == '<div class="agGrid-Markdown"><div><em>Toyota</em> in italics</div></div>'
+    grid.wait_for_cell_text(1, 1, "Mondeo")
+    grid.wait_for_cell_text(2, 2, "Example")
+    assert grid.get_cell(2, 2).get_attribute('innerHTML') == '<div class="agGrid-Markdown"><div><a href="#">Example</a></div></div>'
+    grid.wait_for_cell_text(1, 2, 'Link to new tab')
+    assert grid.get_cell(1, 2).get_attribute('innerHTML') == '<div class="agGrid-Markdown"><div><a href="#" target="_blank">Link to new tab</a></div></div>'
+    assert grid.get_cell(2, 3).get_attribute('innerHTML') == '<div class="agGrid-Markdown">' \
+                                                             '<div><img src="https://www.ag-g' \
+                                                             'rid.com/example-assets/weather/rain.png" ' \
+                                                             'alt="alt text: rain"></div></div>'
```

Comment:
I love the test with the same data displayed both safe and dangerous! It would be clearer though - and less repetitious - to make two separate `Grid` instances like `safe_grid` and `dangerous_grid`, then collect the matching and different items:
```py
for grid in [safe_grid, dangerous_grid]:
    # stuff that should be the same between the two
    grid.wait_for_cell_text(0, 0, "Toyota in italics")

# then cell-by-cell the things that should be different
safe_grid.wait_for_cell_text(1, 2, '<a href="#" target="_blank">Link to new tab</a>')
dangerous_grid.wait_for_cell_text(1, 2, 'Link to new tab')
assert dangerous_grid.get_cell(1, 2).get_attribute('innerHTML') == '<div class="agGrid-Markdown"><div><a href="#" target="_blank">Link to new tab</a></div></div>'
```

Reply:
Alright, updated here: https://github.com/plotly/dash-ag-grid/pull/45/commits/eebe6afbdfab1a2177c9642246084edc7ab1d36f